pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -635,14 +717,19 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 	if provResult.Dirty {
 		info.postSaveCallbacks = append(info.postSaveCallbacks, func() {
 			metricLabels := hostMetricLabels(info.request)
-			if info.host.Spec.Online {
+			if desiredPowerOnState {
 				metricLabels[labelPowerOnOff] = "on"
 			} else {
 				metricLabels[labelPowerOnOff] = "off"
 			}
 			powerChangeAttempts.With(metricLabels).Inc()
 		})
 		info.host.ClearError()
+		if desiredPowerOnState {
+			info.host.RecordPoweredOn()
```

Comment:
This should be recorded and written to the datastore before we ever call `prov.PowerOn()`.
Also, as an added wrinkle, this function is called from both the Ready and Provisioned states, but we only want to record the boot time in the Provisioned state.